dns-server/tests/basic_test.rs

```diff
@@ -85,6 +95,52 @@ pub async fn aaaa_crud() -> Result<(), anyhow::Error> {
     Ok(())
 }
 
+#[tokio::test]
+pub async fn answers_match_question() -> Result<(), anyhow::Error> {
+    let test_ctx = init_client_server("not_wrong_record_answers").await?;
+    let client = &test_ctx.client;
+
+    // records should initially be empty
+    let records = dns_records_list(client, TEST_ZONE).await?;
+    assert!(records.is_empty());
+
+    // add an aaaa record
+    let name = "devron".to_string();
+    let addr = Ipv6Addr::new(0xfd, 0, 0, 0, 0, 0, 0, 0x1);
+    let aaaa = DnsRecord::Aaaa(addr);
+    let input_records = HashMap::from([(name.clone(), vec![aaaa])]);
+    dns_records_create(client, TEST_ZONE, input_records.clone()).await?;
+
+    // read back the aaaa record
+    let records = dns_records_list(client, TEST_ZONE).await?;
+    assert_eq!(input_records, records);
+
+    let name = Name::from_ascii(&(name + "." + TEST_ZONE + "."))
+        .expect("can construct name for query");
+
+    // it turns out `trust-dns-resolver`'s internal CachingClient will
+    // transparently correct the server misbehavior "server sent AAAA answers to
+    // an A query": the caching client will cache the extra record, then
+    // determine that there are no A records to respond with, and send a
+    // matching response.
```

Comment:
What do you mean by "matching response" here?  A response indicating no records?

Does this comment mean that even if we'd had this test before, it would have passed?

Reply:
"matching response" is an attempt to nod to the language a reader might find in RFC 1034 - in this case, yes, a response with all zero records matching the query. re-reading the comment, i could have phrased that better...

about the comment generally: if the test hinged on asserting that `resolver.ipv4_lookup(name)` returned no records when we've added only an AAAA record, it would pass on main. the test as written here, avoiding `CachingClient` would fail on main. (this was very confusing to discover, as i started by using `ipv4_lookup` and realizing i couldn't make this test fail as expected..)